ERROR during swt compilation

<!-- This is a bug report template. By following the instructions below and filling out the sections with your information, you will help the developers get all the necessary data to fix your issue.
You can also preview your report before submitting it. You may remove sections that aren't relevant to your particular case.

Let's begin with a checklist: Replace the empty checkboxes [ ] below with checked ones [x] accordingly. -->

- [X] I have searched open and closed issues for duplicates
- [X] I am submitting a bug report for existing functionality that does not work as intended

----------------------------------------

### Bug description

Building of `swt` package to the PKGBUILD current version or the latest one generates errors during building at `ant -f build-swt.xml compile` line. Here the build output on the latest version:
```
Buildfile: /tmp/makepkg/swt/src/swt-4.31/build-swt.xml

init:
   [tstamp] Honouring environment variable SOURCE_DATE_EPOCH which has been set to 1710852667

compile:
    [javac] Compiling 452 source files to /tmp/makepkg/swt/src/swt-4.31/build
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/browser/WebBrowser.java:190: error: ')' expected
    [javac] 		if (arguments[0] instanceof String string) {
    [javac] 		                                 ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/browser/WebBrowser.java:190: error: not a statement
    [javac] 		if (arguments[0] instanceof String string) {
    [javac] 		                                  ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/browser/WebBrowser.java:190: error: ';' expected
    [javac] 		if (arguments[0] instanceof String string) {
    [javac] 		                                        ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/custom/ScrolledComposite.java:173: error: ')' expected
    [javac] 			} else if (event.widget instanceof Control control) {
    [javac] 			                                         ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/custom/ScrolledComposite.java:173: error: not a statement
    [javac] 			} else if (event.widget instanceof Control control) {
    [javac] 			                                          ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/custom/ScrolledComposite.java:173: error: ';' expected
    [javac] 			} else if (event.widget instanceof Control control) {
    [javac] 			                                                 ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/custom/StyleRange.java:112: error: ')' expected
    [javac] 	if (object instanceof StyleRange style) {
    [javac] 	                               ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/custom/StyleRange.java:112: error: not a statement
    [javac] 	if (object instanceof StyleRange style) {
    [javac] 	                                ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/custom/StyleRange.java:112: error: ';' expected
    [javac] 	if (object instanceof StyleRange style) {
    [javac] 	                                     ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/GlyphMetrics.java:102: error: ')' expected
    [javac] 	if (!(object instanceof GlyphMetrics metrics)) return false;
    [javac] 	                                   ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/GlyphMetrics.java:102: error: illegal start of expression
    [javac] 	if (!(object instanceof GlyphMetrics metrics)) return false;
    [javac] 	                                           ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/GlyphMetrics.java:102: error: ';' expected
    [javac] 	if (!(object instanceof GlyphMetrics metrics)) return false;
    [javac] 	                                            ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1074: error: illegal start of expression
    [javac] 	return switch (transparencyType) {
    [javac] 	      ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1074: error: not a statement
    [javac] 	return switch (transparencyType) {
    [javac] 	             ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1074: error: ';' expected
    [javac] 	return switch (transparencyType) {
    [javac] 	                               ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1075: error: orphaned case
    [javac] 	case SWT.TRANSPARENCY_ALPHA -> getTransparencyMaskFromAlphaData();
    [javac] 	^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1075: error: : expected
    [javac] 	case SWT.TRANSPARENCY_ALPHA -> getTransparencyMaskFromAlphaData();
    [javac] 	                          ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1075: error: illegal start of expression
    [javac] 	case SWT.TRANSPARENCY_ALPHA -> getTransparencyMaskFromAlphaData();
    [javac] 	                           ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1075: error: ';' expected
    [javac] 	case SWT.TRANSPARENCY_ALPHA -> getTransparencyMaskFromAlphaData();
    [javac] 	                                                              ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1076: error: : expected
    [javac] 	case SWT.TRANSPARENCY_MASK -> new ImageData(width, height, 1, bwPalette(), maskPad, maskData);
    [javac] 	                         ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1076: error: illegal start of expression
    [javac] 	case SWT.TRANSPARENCY_MASK -> new ImageData(width, height, 1, bwPalette(), maskPad, maskData);
    [javac] 	                          ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1076: error: ';' expected
    [javac] 	case SWT.TRANSPARENCY_MASK -> new ImageData(width, height, 1, bwPalette(), maskPad, maskData);
    [javac] 	                            ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1077: error: : expected
    [javac] 	case SWT.TRANSPARENCY_PIXEL -> colorMaskImage(transparentPixel);
    [javac] 	                          ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1077: error: illegal start of expression
    [javac] 	case SWT.TRANSPARENCY_PIXEL -> colorMaskImage(transparentPixel);
    [javac] 	                           ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1077: error: ';' expected
    [javac] 	case SWT.TRANSPARENCY_PIXEL -> colorMaskImage(transparentPixel);
    [javac] 	                                            ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1077: error: not a statement
    [javac] 	case SWT.TRANSPARENCY_PIXEL -> colorMaskImage(transparentPixel);
    [javac] 	                                             ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1077: error: ';' expected
    [javac] 	case SWT.TRANSPARENCY_PIXEL -> colorMaskImage(transparentPixel);
    [javac] 	                                                             ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1078: error: : expected
    [javac] 	default -> colorMaskImage(transparentPixel);
    [javac] 	      ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1078: error: illegal start of expression
    [javac] 	default -> colorMaskImage(transparentPixel);
    [javac] 	       ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1078: error: ';' expected
    [javac] 	default -> colorMaskImage(transparentPixel);
    [javac] 	                        ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1078: error: not a statement
    [javac] 	default -> colorMaskImage(transparentPixel);
    [javac] 	                         ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/ImageData.java:1078: error: ';' expected
    [javac] 	default -> colorMaskImage(transparentPixel);
    [javac] 	                                         ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/LineAttributes.java:139: error: ')' expected
    [javac] 	if (!(object instanceof LineAttributes p)) return false;
    [javac] 	                                     ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/LineAttributes.java:139: error: illegal start of expression
    [javac] 	if (!(object instanceof LineAttributes p)) return false;
    [javac] 	                                       ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/LineAttributes.java:139: error: ';' expected
    [javac] 	if (!(object instanceof LineAttributes p)) return false;
    [javac] 	                                        ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/Point.java:82: error: ')' expected
    [javac] 	if (!(object instanceof Point p)) return false;
    [javac] 	                            ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/Point.java:82: error: illegal start of expression
    [javac] 	if (!(object instanceof Point p)) return false;
    [javac] 	                              ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/Point.java:82: error: ';' expected
    [javac] 	if (!(object instanceof Point p)) return false;
    [javac] 	                               ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:117: error: illegal start of expression
    [javac] 		b = switch (i) {
    [javac] 		   ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:117: error: not a statement
    [javac] 		b = switch (i) {
    [javac] 		          ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:117: error: ';' expected
    [javac] 		b = switch (i) {
    [javac] 		             ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:118: error: orphaned case
    [javac] 		case 0 -> {
    [javac] 		^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:118: error: : expected
    [javac] 		case 0 -> {
    [javac] 		     ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:118: error: illegal start of expression
    [javac] 		case 0 -> {
    [javac] 		      ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:118: error: ';' expected
    [javac] 		case 0 -> {
    [javac] 		        ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:123: error: orphaned case
    [javac] 		case 1 -> {
    [javac] 		^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:123: error: : expected
    [javac] 		case 1 -> {
    [javac] 		     ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:123: error: illegal start of expression
    [javac] 		case 1 -> {
    [javac] 		      ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:123: error: ';' expected
    [javac] 		case 1 -> {
    [javac] 		        ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:128: error: orphaned case
    [javac] 		case 2 -> {
    [javac] 		^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:128: error: : expected
    [javac] 		case 2 -> {
    [javac] 		     ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:128: error: illegal start of expression
    [javac] 		case 2 -> {
    [javac] 		      ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:128: error: ';' expected
    [javac] 		case 2 -> {
    [javac] 		        ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:133: error: illegal start of type
    [javac] 		case 3 -> {
    [javac] 		^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:133: error: ';' expected
    [javac] 		case 3 -> {
    [javac] 		   ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:133: error: illegal start of type
    [javac] 		case 3 -> {
    [javac] 		      ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:133: error: <identifier> expected
    [javac] 		case 3 -> {
    [javac] 		        ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:133: error: ';' expected
    [javac] 		case 3 -> {
    [javac] 		          ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:134: error: illegal start of type
    [javac] 			r = p;
    [javac] 			 ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:135: error: <identifier> expected
    [javac] 			g = q;
    [javac] 			^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:138: error: class, interface, or enum expected
    [javac] 		case 4 -> {
    [javac] 		^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:140: error: class, interface, or enum expected
    [javac] 			g = p;
    [javac] 			^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:141: error: class, interface, or enum expected
    [javac] 			yield brightness;
    [javac] 			^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:142: error: class, interface, or enum expected
    [javac] 		}
    [javac] 		^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:145: error: class, interface, or enum expected
    [javac] 			g = p;
    [javac] 			^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:146: error: class, interface, or enum expected
    [javac] 			yield q;
    [javac] 			^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:147: error: class, interface, or enum expected
    [javac] 		}
    [javac] 		^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:150: error: class, interface, or enum expected
    [javac] 			g = p;
    [javac] 			^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:151: error: class, interface, or enum expected
    [javac] 			yield q;
    [javac] 			^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:152: error: class, interface, or enum expected
    [javac] 		}
    [javac] 		^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:154: error: class, interface, or enum expected
    [javac] 	}
    [javac] 	^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:156: error: class, interface, or enum expected
    [javac] 	green = (int)(g * 255 + 0.5);
    [javac] 	^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:157: error: class, interface, or enum expected
    [javac] 	blue = (int)(b * 255 + 0.5);
    [javac] 	^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:158: error: class, interface, or enum expected
    [javac] }
    [javac] ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:172: error: class, interface, or enum expected
    [javac] public float[] getHSB() {
    [javac]        ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:174: error: class, interface, or enum expected
    [javac] 	float g = green / 255f;
    [javac] 	^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:175: error: class, interface, or enum expected
    [javac] 	float b = blue / 255f;
    [javac] 	^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:176: error: class, interface, or enum expected
    [javac] 	float max = Math.max(Math.max(r, g), b);
    [javac] 	^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:177: error: class, interface, or enum expected
    [javac] 	float min = Math.min(Math.min(r, g), b);
    [javac] 	^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:178: error: class, interface, or enum expected
    [javac] 	float delta = max - min;
    [javac] 	^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:179: error: class, interface, or enum expected
    [javac] 	float hue = 0;
    [javac] 	^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:180: error: class, interface, or enum expected
    [javac] 	float brightness = max;
    [javac] 	^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:181: error: class, interface, or enum expected
    [javac] 	float saturation = max == 0 ? 0 : (max - min) / max;
    [javac] 	^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:182: error: class, interface, or enum expected
    [javac] 	if (delta != 0) {
    [javac] 	^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:185: error: class, interface, or enum expected
    [javac] 		} else {
    [javac] 		^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:188: error: class, interface, or enum expected
    [javac] 			} else {
    [javac] 			^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:190: error: class, interface, or enum expected
    [javac] 			}
    [javac] 			^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:193: error: class, interface, or enum expected
    [javac] 		if (hue < 0) hue += 360;
    [javac] 		^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:194: error: class, interface, or enum expected
    [javac] 	}
    [javac] 	^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:196: error: class, interface, or enum expected
    [javac] }
    [javac] ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:209: error: class, interface, or enum expected
    [javac] public boolean equals(Object object) {
    [javac]        ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:211: error: class, interface, or enum expected
    [javac] 	if (!(object instanceof RGB rgb)) return false;
    [javac] 	^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:212: error: class, interface, or enum expected
    [javac] 	return (rgb.red == this.red) && (rgb.green == this.green) && (rgb.blue == this.blue);
    [javac] 	^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:213: error: class, interface, or enum expected
    [javac] }
    [javac] ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:226: error: class, interface, or enum expected
    [javac] public int hashCode() {
    [javac]        ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:228: error: class, interface, or enum expected
    [javac] }
    [javac] ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:237: error: class, interface, or enum expected
    [javac] public String toString() {
    [javac]        ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGB.java:239: error: class, interface, or enum expected
    [javac] }
    [javac] ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGBA.java:128: error: ')' expected
    [javac] 	if (!(object instanceof RGBA rgba)) return false;
    [javac] 	                           ^
    [javac] /tmp/makepkg/swt/src/swt-4.31/org/eclipse/swt/graphics/RGBA.java:128: error: illegal start of expression
    [javac] 	if (!(object instanceof RGBA rgba)) return false;
    [javac] 	                                ^
    [javac] 100 errors
```

### Steps to reproduce

- Going to swt package in BlackArch
- `makepkg -src`

--- a/packages/swt/PKGBUILD
+++ b/packages/swt/PKGBUILD
@@ -8,35 +8,40 @@
 # Contributor: Guillaume ALAUX <guillaume@archlinux.org>
 
 pkgname=swt
-pkgver=4.22
+pkgver=4.31
 pkgrel=1
-_date=202111241800
+_date=202402290520
 pkgdesc='An open source widget toolkit for Java.'
 arch=('x86_64')
 url='https://www.eclipse.org/swt/'
 license=('EPL')
-depends=('java-runtime>=8' 'libxtst')
-makedepends=('ant' 'glu' 'gtk3' 'gtk4' 'jdk8-openjdk' 'pkgconfig' 'unzip'
-             'webkit2gtk')
+depends=('java-runtime>=17' 'libxtst')
+makedepends=('ant' 'glu' 'gtk3' 'gtk4' 'pkgconfig' 'unzip' 'webkit2gtk')
 optdepends=('glu' 'gtk3' 'gtk4')
 _archive="$pkgname-$pkgver-gtk-linux-$arch.zip"
 source=("https://download.eclipse.org/eclipse/downloads/drops4/R-$pkgver-$_date/$_archive"
         'build-swt.xml')
 noextract=("$_archive")
-sha512sums=('cfe56e30d5ec3d9e56e9721a415522edd075456fd37140e35238b75b09e80209417ca9aa46fdc4f797b5e37c4646c6029a9efa8267ce038cc78242015ce9e9a5'
-            '18660eb15b09d70cf0a8b567ea55e4de1c20d9d463908e4d4bb1eba0139a0ba4c9af63e825cc36c286701eca604d4a1f239fa676de5a48ddad7ef9b2c368e543')
+sha512sums=('da61610ce00fc46893ec3ee6d5d7ed5cc59312002c6946a3748cff1a776af75bad0026512b412ef1622ca0bee124764ed8ee7f5f87364cff945b296feb86e74b'
+            '4fb681c8ff716fc7223331bae98b8c8c2ae1d7f7a9738b210f4ca673f0995b9f0463c97b40d84a2db6b4d3f8f583cb3baa05c4694e796b0e1e32e831fdda3116')
 
 prepare() {
   unzip -oq "$_archive" -d "$pkgname-$pkgver"
+
+  cd "$pkgname-$pkgver"
+
+  unzip -oq src.zip
+
+  sed -i 's/-fPIC/& -Wno-error=deprecated-declarations/' library/build.sh
 }
 
 build() {
   cd "$pkgname-$pkgver"
 
-  unzip -oq src.zip
   cp $srcdir/build-swt.xml .
 
-  export JAVA_HOME=/usr/lib/jvm/java-8-openjdk
+  export JAVA_HOME=/usr/lib/jvm/$(archlinux-java get)
+  export SWT_JAVA_HOME=${JAVA_HOME}
 
   ant -f build-swt.xml compile
   ant -f build-swt.xml jar
@@ -51,6 +56,6 @@ package() {
 
   # Shared objects
   install -d "$pkgdir/usr/lib"
-  OUTPUT_DIR="$pkgdir/usr/lib" ./build.sh -gtk-all install
+  OUTPUT_DIR="$pkgdir/usr/lib" ./library/build.sh -gtk-all install
 }
 

--- a/packages/swt/build-swt.xml
+++ b/packages/swt/build-swt.xml
@@ -8,7 +8,7 @@
    </target>
 
    <target name="compile" depends="init" description="Compile the SWT toolset">
-      <javac srcdir="${src}" destdir="${build}" source="1.8" includeantruntime="false"/>
+      <javac srcdir="${src}" destdir="${build}" source="17" includeantruntime="false"/>
    </target>
 
    <target name="jar">

